README.md

```diff
@@ -0,0 +1,289 @@
+┌──────────────┐                    │   Device Service   │               ┌─────────────────┐
+│              │                    │                    │               │                 │
+│              │ REST response      │                    │ SOAP response │                 │
+│    Client  ◄─┼────────────────────┼───  Transform   ◄──┼───────────────┼── Onvif Camera  │
+│              │                    │   to REST response │               │                 │
+│              │                    │                    │               │                 │
+└──────────────┘                    └────────────────────┘               └─────────────────┘
+```
+
+## General Usage
+
+### Run Unit Test
+```shell
+make test
+```
+
+### Build the executable file
+```shell
+make build
+```
+
+### Build docker image
+Build docker image named edgex/device-onvif-camera:0.0.0-dev with the following command:
+```shell
+make docker
+```
+
+### Define the device profile
+
+The device resource should provide two attributes:
+* **service** indicates the web service for the Onvif
+* **function** indicates the SOAP action for the specified web service
+
+For example:
+```yaml
+deviceResources:
+  - name: "Hostname"
+    isHidden: false
+    description: "Camera Hostname"
+    attributes:
+      service: "Device"
+      getFunction: "GetHostname"
+      setFunction: "SetHostname"
+    properties:
+      valueType: "Object"
+      readWrite: "RW"
+```
+
+See the sample at [cmd/res/profiles/camera.yaml](cmd/res/profiles/camera.yaml)
+
+### Define the device
+
+The device's protocol properties should contain:
+* **Address** is the IP address of the Onvif camera
+* **Port** is the server port of the Onvif camera
+* **AuthMode** indicates the auth mode of the Onvif camera
+* **SecretPath** indicates the path to retrieve the username and password
+
+For example:
+```yaml
+[[DeviceList]]
+Name = "Camera001"
+ProfileName = "camera"
+Description = "My test camera"
+  [DeviceList.Protocols]
+    [DeviceList.Protocols.Onvif]
+    Address = "192.168.12.123"
+    Port = 80
+    # Assign AuthMode to "digest" | "usernametoken" | "both" | "none"
+    AuthMode = "usernametoken"
+    SecretPath = "credentials001"
+```
+See the sample at [cmd/res/devices/camera.toml.example](cmd/res/devices/camera.toml.example)
+
+
+## Getting Started Guide
+- [Getting started guide for developer](./doc/getting-started-guide.md)
+- [Getting started guide for running docker container with none security mode](./doc/getting-started-with-docker.md)
+- [Getting started guide for running docker container with security mode](./doc/getting-started-with-docker-security.md)
+
+## Tested Onvif Camera
+[Tested Onvif cameras with Onvif functions.](./doc/tested-onvif-camera.md)
+
+## Pass parameter via URL query parameter
+[Get Command Parameter](./doc/get-cmd-parameter.md)
+
+## Custom feature for device-onvif-go
+[RebootNeeded](./doc/custom-feature-rebootneeded.md)
+
+## API Usage
+
```

Comment:
Consider bulletized list, since items (User Handler, Analytics Support) show up in one link.

Reply:
Modified to use the bulletized list.